app/gui/enso-profiler-enso-data/src/beanpole.rs

```diff
@@ -0,0 +1,162 @@
+//! A diagram type for illustrating timings of messages between processes.
+
+
+
+// ===============
+// === Diagram ===
+// ===============
+
+/// The data necessary to create a diagram of message timings.
+#[derive(Debug, Default)]
+pub struct Diagram<'a> {
+    processes: Vec<&'a str>,
+    messages:  Vec<Message>,
+}
+
+impl<'a> Diagram<'a> {
+    /// Define a new process. Return a handle.
+    pub fn process<'b: 'a>(&mut self, name: &'b str) -> Process {
+        let id = self.processes.len();
+        self.processes.push(name);
+        Process(id)
+    }
+
+    /// Log a message between two processes.
+    pub fn message(&mut self, p0: Process, p1: Process, time: f64, label: String) {
+        self.messages.push(Message { p0, p1, time, label });
+    }
+}
+
+
+// === Process ===
+
+/// A process that may send and receive messages.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct Process(usize);
```

Comment:
I understand that the `size` is the process id. If so, accessing it as `process.0` is not explanatory. Let's use `process.id` instead.